src/cart/src/cart/crt_iv.c

```diff
@@ -3149,6 +3156,13 @@ crt_iv_update_internal(crt_iv_namespace_t ivns, uint32_t class_id,
 		IVNS_ADDREF(ivns_internal);
 
 		cb_info->uci_class_id = class_id;
+		if (ivns_internal->cii_grp_priv->gp_self == CRT_NO_RANK) {
+			IV_DBG(iv_key, "%s: self rank not known yet\n",
+			       ivns_internal->cii_grp_priv->gp_pub.cg_grpid);
+			IVNS_DECREF(cb_info->uci_ivns_internal);
+			D_FREE_PTR(cb_info);
+			D_GOTO(put, rc = -DER_GRPVER);
+		}
```

Comment:
any reason why not do this check earlier on line 3098 for example?

Reply:
No strong reasons. I was just not sure if any of the callbacks above would want to update the self rank (or would reschedule to other ULTs who might want to update the self rank). (Sounds unlikely.) Hence, I picked a more conservative location for this check. If it's better to move it, I have no problem to move it at all. Let me know.

Reply:
i think it would be safer to move it at the top of the function as ivns_internal->cii_grp_priv->gp_self is being used in few places in this function above, and being CRT_NO_RANK might cause some strange side-effects in how requests get propagated.

I think we can safely restrict update of gp_self to be only done as part of pre-forward callbacks of corpcs and any other place where check for gp_self is done should short circuit early with DER_GPVER

Reply:
Moved to line 3098.

I had considered the earlier usages of gp_self in this function. They are okay if gp_self == CRT_NO_RANK, I think.